ui/webui/test/check-storage

```diff
@@ -735,13 +745,73 @@ class TestStorageMountPoints(anacondalib.VirtInstallMachineCase):
         i.next()
 
         # verify review screen
-        disk = "vda"
-        r.check_disk(disk, "16.1 GB vda (0x1af4)")
+        r.check_disk(dev, "16.1 GB vda (0x1af4)")
 
-        r.check_disk_row(disk, 1, "vda1, 1.05 MB: biosboot")
-        r.check_disk_row(disk, 2, "vda2, 1.07 GB: format as ext4, /boot")
-        r.check_disk_row(disk, 3, "root, 15.0 GB: format as btrfs, /")
-        r.check_disk_row(disk, 4, "home, 15.0 GB: mount, /home")
+        r.check_disk_row(dev, 1, "vda1, 1.05 MB: biosboot")
+        r.check_disk_row(dev, 2, "vda2, 1.07 GB: format as ext4, /boot")
+        r.check_disk_row(dev, 3, "root, 15.0 GB: format as btrfs, /")
+        r.check_disk_row(dev, 4, "home, 15.0 GB: mount, /home")
+
+        i.back(previous_page=i.steps.CUSTOM_MOUNT_POINT)
+        i.back()
+
+        # Checks for nested btrfs subvolume
+        tmp_mount = "/tmp/btrfs-mount-test"
+        m.execute(f"""
+        mkdir -p {tmp_mount}
+        mount {disk}3 {tmp_mount}
+        btrfs subvolume create {tmp_mount}/home/Movies
+        btrfs subvolume create {tmp_mount}/home/Movies/Good_Movies
+        btrfs subvolume create {tmp_mount}/home/Movies/Bad_Movies
+        umount {tmp_mount}
+        rmdir {tmp_mount}
+        """)
+        s.rescan_disks()
+        self.select_mountpoint(i, s, [(dev, True)])
+
+        self.select_row_device(1, f"{dev}2")
+        self.select_row_device(2, "root")
+        self.add_mount()
+        self.select_row_device(3, "home")
+        self.select_row_mountpoint(3, "/home")
+        self.add_mount()
+        self.select_row_device(4, "home/Movies")
+        self.select_row_mountpoint(4, "/home/Movies")
+        self.add_mount()
+        self.select_row_device(5, "home/Movies/Good_Movies")
+        self.select_row_mountpoint(5, "/home/Movies/Good_Movies")
+        self.add_mount()
+        self.select_row_device(6, "home/Movies/Bad_Movies")
+        self.select_row_mountpoint(6, "/home/Movies/Bad_Movies")
+
+        # No error when no devices are reformatted
+        for row in range(3, 6):
+            self.wait_mount_point_table_column_helper(row, "format", present=False)
+
+        # When parent is re-formatted all child devices must be reformatted
+        self.select_row_device(4, "home/Movies")
+        self.select_reformat(4)
+        self.wait_mount_point_table_column_helper(4, "format", text="Mismatch")
+        self.select_reformat(5)
+        self.select_reformat(6)
+        self.wait_mount_point_table_column_helper(4, "format", present=False)
+
+        # When parent is re-formatted all child devices included nested cannot be reformatted
+        self.select_reformat(3)
+        self.wait_mount_point_table_column_helper(3, "format", present=False)
+        self.select_reformat(6, False)
+        self.wait_mount_point_table_column_helper(3, "format", text="Mismatch")
+
+        # When parent is re-formmated all child devices should be selected (part of the mountpoint assignment table)
```

Comment:
So this is expected behaviour (I have two more nested subvolumes `a/b` and `a/b/c`?
![image](https://github.com/rhinstaller/anaconda/assets/5063197/b6c9f336-eb36-4a75-ab4e-d5a77a336b76)

I don't think this should be allowed -- the nested subvolumes `a/b` and `a/b/c` will be removed, but that should be a supported use case. When reinstalling Fedora, we'll have a subvolume under root (`root/var/lib/machines` created automatically by systemd) and we should allow to just remove it by reformatting `root` without assigning a mount point to it.

Reply:
Fixed.